Libs/vtkSegmentationCore/vtkOrientedImageDataResample.cxx

```diff
@@ -1835,9 +1835,100 @@ bool vtkOrientedImageDataResample::IsLabelInMask(
   return valueFound;
 }
 
+//----------------------------------------------------------------------------
+bool vtkOrientedImageDataResample::ValidateSegmentationImageType(vtkImageData* image)
+{
+  if (!image)
+  {
+    vtkErrorWithObjectMacro(nullptr, "ValidateSegmentationImageType: Invalid input image");
+    return false;
+  }
+
+  vtkIdType validScalarTypes[3] =
+  {
+    VTK_UNSIGNED_CHAR,
+    VTK_UNSIGNED_SHORT,
+    VTK_UNSIGNED_INT
+    //VTK_UNSIGNED_LONG // See issue #5427.
+                        // vtkImageThreshold is not able to fill thresholded images with VTK_LONG_MAX, preventing segment erase.
+  };
+
+  int scalarType = image->GetScalarType();
+  for (int i = 0; i < 3; i++)
+  {
+    if (scalarType == validScalarTypes[i])
+    {
+      return true;
+    }
+  }
+
+  double scalarRange[2] = { -1.0, 0.0 };
+  image->GetScalarRange(scalarRange);
+  if (scalarRange[0] < 0)
+  {
+    vtkErrorWithObjectMacro(image, "ValidateSegmentationImageType: Negative values are present, so image cannot be cast to unsigned type");
+    return false;
+  }
+
+  vtkGenericWarningMacro("ValidateSegmentationImageType: Invalid scalar type (" << image->GetScalarTypeAsString()
+    << "), converting image to smallest unsigned integer type");
+
+  // Attempt to cast the image to a valid scalar type.
+  return vtkOrientedImageDataResample::CastToSmallestUnsignedIntegerType(image);
+}
 
 //----------------------------------------------------------------------------
-void vtkOrientedImageDataResample::CastImageForValue(vtkOrientedImageData* image, double value)
+bool vtkOrientedImageDataResample::CastToSmallestUnsignedIntegerType(vtkImageData* image)
+{
+  if (!image)
+  {
+    vtkErrorWithObjectMacro(nullptr, "CastImageToSmallestIntegerType: Invalid input image");
+    return false;
+  }
+
+  double scalarRange[2];
+  image->GetScalarRange(scalarRange);
+  if (scalarRange[0] < 0)
+  {
+    vtkErrorWithObjectMacro(image, "CastImageToSmallestIntegerType: Negative values are present, so image cannot be cast to unsigned type");
+    return false;
+  }
+
+  int scalarType = image->GetScalarType();
+  if (scalarRange[1] <= VTK_UNSIGNED_CHAR_MAX)
+  {
+    scalarType = VTK_UNSIGNED_CHAR;
+  }
+  else if (scalarRange[1] <= VTK_UNSIGNED_SHORT_MAX)
+  {
+    scalarType = VTK_UNSIGNED_SHORT;
+  }
+  else if (scalarRange[1] <= VTK_UNSIGNED_INT_MAX)
+  {
+    scalarType = VTK_UNSIGNED_INT;
+  }
+  else
+  {
+    vtkErrorWithObjectMacro(image, "CastImageToSmallestIntegerType: Scalar range is too large to fit into an integer type");
```

Comment:
We could enable `ClampOverflow` in `vtkImageCast` and make this a warning rather than an error.  It might be more useful to the user to have something load rather than fail.  Or make it an option if we don't want to lose data with only a warning.

Reply:
Ok, we now clamp overflow and log a warning.